### DISCUSSION TITLE
Option for the picture to open in the same window

### DISCUSSION THREAD

**Discussion by olympia** [algertc/ALPR-Database#19] [Ideas]
Could you add an option for pictures to open in the same window, allowing us to use the browser's back button instead of closing the new window they currently open in?

**Comment by algertc** [comment]
I'll probably convert this to a modal. I've been meaning to but haven't gotten to it. I agree the open raw in new tab is kind of disjointed.

I'll add some buttons to download and open in new tab to the modal element.


I really need to refactor to not store the images as base64 in a database column though 😭. Incredibly jank and horrendously inefficient. Repo created <30 days ago and already has tech debt. I was going for a quick and easy solution just to get something going when I originally had the idea for this, so I tolerated it. Here we are and I still haven't changed it. Going to be rough to transition that.

**Comment by olympia** [comment]
Modal would be indeed an even more polished solution, but I think having the pic opening in the same window would be already a relief in many cases :)

As for the rest, personally I wouldn't mind to lose the Db pictures in case the migration/ transition is too difficult.

What you did is already very very very useful.

**Comment by olympia** [comment]
Thank you for providing a solution for this as well! Working great!

Only one thing what might make it (at least for me) even more convenient and probably it's not a big thing: 
would it be possible to also close the model if we click on the picture itself?

**Reply by algertc** [reply]
Yes, but I think that would be confusing UX for most people. Clicking anywhere outside of the modal or hitting escape will close it.

**Comment by olympia** [comment]
Blue Iris WebIU does the same with camera windows. It's very convenient. I thought it's a good best practice :)
